components/org.wso2.carbon.identity.api.server.api.resource/org.wso2.carbon.identity.api.server.api.resource.v1/src/main/java/org/wso2/carbon/identity/api/server/api/resource/v1/core/ServerAPIResourceManagementService.java

```diff
@@ -536,10 +536,19 @@ private void validateRequiredAttributes(List<String> requiredAttributeList) thro
      */
     private void handleSystemAPI(APIResource apiResource) {
 
-        if (apiResource.getType() != null &&
-                apiResource.getType().startsWith(APIResourceMgtEndpointConstants.SYSTEM_API_RESOURCE_TYPE)) {
+        if (apiResource.getType() == null) {
+            return;
+        }
+        if (apiResource.getType().startsWith(APIResourceMgtEndpointConstants.SYSTEM_API_RESOURCE_TYPE) ||
```

Comment:
Shall we check other way around whe.ther it is not BUSINESS type? In that case, newly added types will be not modifiable by default